File: backend/Models/JobPost.cs
namespace LocalHire.Api.Models;

public sealed class JobPost
{
    public Guid Id { get; set; }
    public Guid EmployerId { get; set; }
    public UserRole EmployerRole { get; set; } = UserRole.Hiring;
    public required string Title { get; set; }
    public required string Description { get; set; }
    public required string WorkplaceName { get; set; }
    public required string CityArea { get; set; }
    public string? State { get; set; }
    public string? Pincode { get; set; }
    public double? Latitude { get; set; }
    public double? Longitude { get; set; }

    // Role details
    public EmploymentType? EmploymentType { get; set; }
    public decimal? SalaryMin { get; set; }
    public decimal? SalaryMax { get; set; }
    public SalaryPeriod? SalaryPeriod { get; set; }
    public string? MinEducation { get; set; }
    public int? ExperienceMinYears { get; set; }
    public int? ExperienceMaxYears { get; set; }
    public string? WorkingDays { get; set; }
    public TimeOnly? ShiftStartTime { get; set; }
    public TimeOnly? ShiftEndTime { get; set; }
    public int? Openings { get; set; }
    public List<string> RequiredSkills { get; set; } = new();
    public List<string> Languages { get; set; } = new();
    public List<string> Benefits { get; set; } = new();

    public bool IsActive { get; set; } = true;
    public DateTimeOffset CreatedAt { get; set; }

    public User Employer { get; set; } = null!;
    public ICollection<JobApplication> Applications { get; set; } = new List<JobApplication>();
}

File: integration-tests/features/db_models/job_post.py
"""Row model for the ``JobPosts`` table (see ``backend/Models/JobPost.cs``)."""

from dataclasses import dataclass, field
from datetime import datetime
from decimal import Decimal
from typing import Optional

from features.db_models.model import Model
from features.utils.util import utc_now


@dataclass
class JobPost(Model):
    id: str
    employer_id: str
    title: str
    description: str
    workplace_name: str
    city_area: str
    employer_role: str = "Hiring"
    state: Optional[str] = None
    pincode: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    employment_type: Optional[str] = None  # FullTime | PartTime | Contract | Temporary
    salary_min: Optional[Decimal] = None
    salary_max: Optional[Decimal] = None
    salary_period: Optional[str] = None  # Hourly | Daily | Weekly | Monthly | Yearly
    min_education: Optional[str] = None
    experience_min_years: Optional[int] = None
    experience_max_years: Optional[int] = None
    working_days: Optional[str] = None
    shift_start_time: Optional[str] = None  # "HH:MM"
    shift_end_time: Optional[str] = None
    openings: Optional[int] = None
    required_skills: list = field(default_factory=list)
    languages: list = field(default_factory=list)
    benefits: list = field(default_factory=list)
    is_active: bool = True
    created_at: datetime = field(default_factory=utc_now)
